contrib/babelfishpg_tsql/src/pl_handler.c

```diff
@@ -6391,6 +6584,681 @@ bbf_ExecDropStmt(DropStmt *stmt)
+	List		*res;
+	StringInfoData	query;
+
+	/*
+	 * We prepare the following query to CREATE PARTITION of partitioned table.
+	 * This will be executed using ProcessUtility().
+	 */
+	initStringInfo(&query);
+	appendStringInfo(&query, "CREATE TABLE \"%s\".dummy PARTITION OF \"%s\".\"%s\""
+				" FOR VALUES FROM ('dummy') TO ('dummy');",
+				physical_schema_name, physical_schema_name, relname);
+	PG_TRY();
+	{
+		set_config_option("babelfishpg_tsql.sql_dialect", "postgres", GUC_CONTEXT_CONFIG,
+					PGC_S_SESSION, GUC_ACTION_SAVE, true, 0, false);
+		res = raw_parser(query.data, RAW_PARSE_DEFAULT);
+		partition_stmt = (CreateStmt *) parsetree_nth_stmt(res, 0);
+	}
+	PG_FINALLY();
+	{
+		/* Reset dialect back to original value. */
+		set_config_option("babelfishpg_tsql.sql_dialect", old_dialect, GUC_CONTEXT_CONFIG,
+					PGC_S_SESSION, GUC_ACTION_SAVE, true, 0, false);
+		pfree(query.data);
+	}
+	PG_END_TRY();
+	return partition_stmt;
+}
+
+/*
+ * construct_unique_hash
+ * 	Constructs a unique hash based on the relation name,
+ * 	this is used to construct unique names for partitions.
+ */
+static char*
+construct_unique_hash(char *relation_name)
+{
+	char		*md5;
+	bool		success;
+	const char	*errstr = NULL;
+
+	md5 = (char *) palloc(MD5_HASH_LEN + 1);
+
+	success = pg_md5_hash(relation_name, strlen(relation_name), md5, &errstr);
+	
+	if (unlikely(!success)) /* Out of memory. */
+	{
+		ereport(ERROR,
+				(errcode(ERRCODE_INTERNAL_ERROR),
+				 errmsg("constructing unique hash name failed for relation \"%s\": %s", relation_name, errstr)));
+	}
+
+	return md5;
+}
+
+/*
+ * set_partition_range_bounds
+ * 	This function sets the lower and upper bounds for a range partition based on its index
+ * 	and the total number of partitions. It handles the following cases:
+ * 	1. For the first partition, it sets the lower bound to DEFAULT to
+ * 		accommodate NULL values along with other values.
+ * 	2. For the last partition, it sets the upper bound to MAXVALUE.
+ * 	3. For other partitions, it sets the bounds to the corresponding values in the range_values array.
+ */
+static void
+set_partition_range_bounds(PartitionBoundSpec *partbound, Datum *range_values, int idx,
+				int total_partitions, bool is_binary_datatype)
+{
+	
+	/* Set lower bound of partition. */
+	if (idx == 0) /* first partition */
+	{
+		partbound->is_default = true;
+		partbound->location = -1;
+		partbound->lowerdatums = NIL;
+		partbound->upperdatums = NIL;
+		return;
+	}
+	else
+	{
+		A_Const *node = makeNode(A_Const);
+		set_node_value_from_datum(node, range_values[idx-1], is_binary_datatype);
+		partbound->location = -1;
+		partbound->is_default = false;
+		partbound->lowerdatums = list_make1(node);
+	}
+
+	/* Set upper bound of partition. */
+	if (idx == total_partitions - 1) /* last partition */
+	{
+		ColumnRef *node = makeNode(ColumnRef);
+		node->fields = list_make1(makeString("maxvalue"));
+		partbound->location = -1;
+		partbound->is_default = false;
```

Comment:
Nit: Do we need to explicitly set location/is_default. It was one time handled in lower bound setting right?

Reply:
is_default is needed, removed location in next commit.